### PR TITLE
fix handler initialization if child registration takes too long

### LINKED ISSUE #2559: ThingHandlers stay UNINITIALIZED of registration runs into timeout
The method `ThingManager.registerChildHandlers()` has the following code:

```
                        registerHandler(child);
                        initializeHandler(child);
```
Within `registerHandler` the handler registration is done in a "Safe" call, i.e. it might return after 5 seconds without having successfully registered the handler by that time.
As a result, `initializeHandler` will simply do nothing, since it checks whether the handler is correctly registered already.

The outcome is that the Thing stays in UNINITIALIZED state indefinitely.

[This seems to be like a report](https://community.openhab.org/t/binding-handler-registration-issues/17422) of exactly that situation.

### PR DESCRIPTION
**Pull request by sjka**
fixes #2559
Signed-off-by: Simon Kaufmann <simon.kfm@googlemail.com>

### CODE REVIEW

**Comment by sjka** [comment]
@sbussweiler could you please also have a look?

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java [hunk 1]
```diff
@@ -1020,15 +1008,45 @@ protected void addThingHandlerFactory(ThingHandlerFactory thingHandlerFactory) {
...
+                SafeMethodCaller.call(new SafeMethodCaller.ActionWithException<Void>() {
+                    @Override
+                    public Void call() throws Exception {
+                        registerHandler(thing, thingHandlerFactory);
+                        initializeHandler(thing);
+                        return null;
+                    }
+                });
+            } catch (TimeoutException e) {
+                // TODO Auto-generated catch block
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java]: na, na, na ;-)

File: bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java [hunk 2]
```diff
@@ -1020,15 +1008,45 @@ protected void addThingHandlerFactory(ThingHandlerFactory thingHandlerFactory) {
...
+                e.printStackTrace();
+            } catch (ExecutionException e) {
+                // TODO Auto-generated catch block
```
Comment (kaikreuzer) [inline bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/internal/ThingManager.java]: dito

**Comment by sbussweiler** [comment]
@sjka lgtm, thanks for the fix.

**Comment by sjka** [comment]
I fixed the auto-generated blocks. 

However, I noticed that registerAndInitializeHandler() -> ... -> doInitializeHandler() leads to nested SafeMethodCaller invocations. 
I will have a look tomorrow in order to sort that out, just in case this would block when the pool runs out of threads.

**Pull request merged**